mongo/mongo.go

```diff
@@ -284,16 +284,12 @@ func IsMaster(session *mgo.Session, obj WithAddresses) (bool, error) {
 }
 
 // SelectPeerAddress returns the address to use as the mongo replica set peer
-// address by selecting it from the given addresses. If no addresses are
-// available an empty string is returned.
-func SelectPeerAddress(addrs []network.Address) string {
-	// ScopeMachineLocal addresses are never suitable for mongo peers,
-	// as each controller runs on a separate machine.
-	const allowMachineLocal = false
```

Comment:
This exercise revealed that we never called this method allowing local-machine addresses. The corresponding scope match in `core/network` has been removed.